### ISSUE TITLE
prefect deployment create failure does not return non-zero exit code

### ISSUE THREAD

**Issue by tekumara** [PrefectHQ/prefect#5687]
## Description

When `prefect deployment create` fails it does not return a non-zero exit code and so automation like CI/CD cannot detect the failure.

## Reproduction / Example

```
❯ prefect deployment create k8s_flow.py             
Loading deployment specifications from python script at 'k8s_flow.py'...
Creating deployment 'test-deployment' for flow 'test-flow'...
Deploying flow script from '/Users/tekumara/code/orion-demo/flows/k8s_flow.py' using File Storage...
Traceback (most recent call last):
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/prefect/cli/deployment.py", line 251, in create
    await spec.create_deployment(validate=False)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/prefect/client.py", line 82, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/prefect/deployments.py", line 287, in create_deployment
    storage_token = await self.flow_storage.write(flow_bytes)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/prefect/blocks/storage.py", line 124, in write
    file = fsspec.open(self.base_path + key, "wb", **self.options)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/fsspec/core.py", line 462, in open
    return open_files(
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/fsspec/core.py", line 305, in open_files
    
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/fsspec/core.py", line 305, in <listcomp>
    
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/fsspec/asyn.py", line 85, in wrapper
    return sync(self.loop, func, *args, **kwargs)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/fsspec/asyn.py", line 65, in sync
    raise return_result
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/fsspec/asyn.py", line 25, in _runner
    result[0] = await coro
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/s3fs/core.py", line 767, in _makedirs
    await self._mkdir(path, create_parents=True)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/s3fs/core.py", line 752, in _mkdir
    await self._call_s3("create_bucket", **params)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/s3fs/core.py", line 302, in _call_s3
    raise err
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/s3fs/core.py", line 282, in _call_s3
    out = await method(**additional_kwargs)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/client.py", line 214, in _make_api_call
    http, parsed_response = await self._make_request(
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/client.py", line 234, in _make_request
    return await self._endpoint.make_request(operation_model, request_dict)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/endpoint.py", line 79, in _send_request
    request = await self.create_request(request_dict, operation_model)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/endpoint.py", line 70, in create_request
    await self._event_emitter.emit(event_name, request=request,
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/hooks.py", line 27, in _emit
    response = await handler(**kwargs)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/signers.py", line 16, in handler
    return await self.sign(operation_name, request)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/aiobotocore/signers.py", line 63, in sign
    auth.add_auth(request)
  File "/Users/tekumara/code/orion-demo/.venv/lib/python3.9/site-packages/botocore/auth.py", line 388, in add_auth
    raise NoCredentialsError()
botocore.exceptions.NoCredentialsError: Unable to locate credentials

Failed to create deployment 'test-flow/test-deployment'

❯ echo $?
0
```

**Comment by madkinsz** [comment]
Thanks! This will be fixed in the next release.

**Comment by madkinsz** [comment]
This should be fixed in 2.0b4+
